cmsimple/classes/LinkChecker.php

```diff
@@ -235,34 +252,74 @@ private function checkExternalLink(array $parts)
         if (isset($parts['query'])) {
             $path .= "?" . $parts['query'];
         }
-        $status = $this->makeHeadRequest($parts['host'], $path);
+        $status = $this->makeHeadRequest($parts['scheme'], $parts['host'], $path);
         return ($status !== false) ? $status : Link::STATUS_EXTERNALFAIL;
     }
 
     /**
      * Makes a head request and returns the response status code, FALSE if the
      * request failed.
      *
+     * @param string $scheme http(s).
      * @param string $host A host name.
      * @param string $path An absolute path.
      *
      * @return int|false
      */
-    protected function makeHeadRequest($host, $path)
+    protected function makeHeadRequest($scheme, $host, $path)
     {
-        $errno = $errstr = null;
-        $socket = fsockopen($host, 80, $errno, $errstr, 5);
-        if ($socket) {
-            $request = "HEAD $path HTTP/1.1\r\nHost: $host\r\n"
-                . "User-Agent: CMSimple_XH Link-Checker\r\n\r\n";
-            fwrite($socket, $request);
-            $response = fread($socket, 12);
-            fclose($socket);
-            $status = substr($response, 9);
-            return (int) $status;
-        } else {
-            return false;
+        global $cf;
+
+        $url = $scheme . '://' . $host . $path;
+        $timeout = 5;
+        $maxredir = (int) $cf['link']['redir'];
+
+        if (function_exists('curl_init')) {
```

Comment:
To make it clear that not only `curl_init()` is required, maybe:
```suggestion
        if (extension_loaded('curl')) {
```